src/Listeners/BazaarEnabled.php

```diff
@@ -44,7 +49,51 @@ public function __construct(ExtensionManager $extensions, SettingsRepositoryInte
      */
     public function subscribe(Dispatcher $events)
     {
-        $events->listen(ConfigureWebApp::class, [$this, 'authenticate']);
+        $events->listen(ConfigureWebApp::class, [$this, 'authenticate'], -10);
+        $events->listen(ConfigureWebApp::class, [$this, 'syncLock'], 100);
+    }
+
+    public function syncLock(ConfigureWebApp $event)
+    {
+        $lockPath = base_path('composer.lock');
```

Comment:
This lock reading and sending code is both in `src/Jobs/SyncLock.php` and `src/Listeners/BazaarEnabled.php`, one should probably go.